File: Two-Sum/two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    map = {}
    result = []
    nums.each_with_index do |item, index|
        map[item] = index
    end
    nums.each_with_index do |item, index|
        x = target - item
        if map.has_key?(x)
            index_x = map[x]
            if index_x != index
                result<<index
                result<<index_x
                break
            end
        end
    end
    result
end

File: Two-Sum/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var map = [Int:Int]();
        var result = [Int]();
        for(i, item ) in nums.enumerated(){
            map[item]=i;
        }
        for(j,item) in nums.enumerated(){
            let x = target-item;
            if let index = map[x]{
                if(index != j){
                    result.append(j);
                    result.append(index);
                    break;
                }
            }
        }
        return result;
    }
}
